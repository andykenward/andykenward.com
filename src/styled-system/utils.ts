import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { colorPropsKeys } from "./colors"
import { fontSizePropsKeys } from "./fonts"

const propKeys = [...colorPropsKeys, ...fontSizePropsKeys]

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)
