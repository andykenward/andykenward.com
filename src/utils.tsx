import shouldForwardProp from "@styled-system/should-forward-prop"
import { ThemeProvider } from "emotion-theming"
import { equals, indexOf } from "ramda"
import React from "react"
import { render } from "@testing-library/react"

import { theme } from "./theme"

export const filterProps: (prop: string, propsToFilter: string[]) => boolean = (
  prop: string,
  propsToFilter: string[]
): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propsToFilter), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" || prop === "data-test-id"

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>)
