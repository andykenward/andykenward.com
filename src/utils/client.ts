import { GraphQLClient } from "graphql-request"

import { getSdk } from "../../generated/graphql"

const graphqlClient = new GraphQLClient(process.env.API_URL, {
  headers: {
    authorization: `Bearer ${process.env.API_TOKEN}`,
  },
})

export const appSdk = (() => getSdk(graphqlClient))()
