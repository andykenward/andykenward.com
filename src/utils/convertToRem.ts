import { is } from "ramda"
/**
 *
 * UTILS
 */

export const convertToRem = (n: number | string): string | number =>
  is(Number, n) && n != 0 ? `${(n as number) / 16}rem` : n
