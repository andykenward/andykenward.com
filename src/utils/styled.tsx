import styled, { CreateStyled } from "@emotion/styled"
import React from "react"

import { Theme } from "../theme"

/**
 * TODO: pass `as` typings down if component
 */
export interface StyledProps {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
}

export default styled as CreateStyled<Theme>
