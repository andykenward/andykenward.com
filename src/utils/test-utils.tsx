import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { ThemeProvider } from "theme-ui"

import { theme } from "../theme"

const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"

export { customRender as render }
