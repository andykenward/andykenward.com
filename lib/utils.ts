import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { propKeys } from "./styled-system"

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" ?? prop === "data-test-id"
