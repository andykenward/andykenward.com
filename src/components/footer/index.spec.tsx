import React from "react"

import { renderWithTheme } from "../../utils"
import { SiteSiteMetadataContact, Maybe } from "../../types/graphqlTypes.d"

import { Footer } from "."

const mockData: Maybe<SiteSiteMetadataContact>[] = [
  { title: "title one", url: "url one" },
  { title: "title two", url: "url two" },
  { title: undefined, url: "url three" },
  { title: "title four", url: undefined },
  { title: undefined, url: undefined },
]

describe("<Footer />", () => {
  it("renders null with empty array", () => {
    const { container } = renderWithTheme(<Footer data={[]} />)
    expect(container.firstChild).toBeNull()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders with 2 items", () => {
    const { container } = renderWithTheme(<Footer data={mockData} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
