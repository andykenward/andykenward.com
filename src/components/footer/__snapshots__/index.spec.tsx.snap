// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders null with empty array 1`] = `null`;

exports[`<Footer /> renders with 2 items 1`] = `
.emotion-5 {
  padding-top: 32px;
  padding-bottom: 32px;
  padding-left: 32px;
  padding-right: 32px;
}

@media screen and (min-width:40em) {
  .emotion-5 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media screen and (min-width:40em) {
  .emotion-5 {
    width: 75%;
  }
}

.emotion-4 {
  margin: 0;
  padding: 0;
}

.emotion-1 {
  margin: 0;
  margin-bottom: 16px;
  padding: 0;
  display: list-item;
}

.emotion-0 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
}

<footer
  class="emotion-5"
>
  <ul
    class="emotion-4"
  >
    <li
      class="emotion-1"
    >
      <a
        class="emotion-0"
        href="url one"
        rel="noopener noreferrer"
        target="_blank"
      >
        title one
      </a>
    </li>
    <li
      class="emotion-1"
    >
      <a
        class="emotion-0"
        href="url two"
        rel="noopener noreferrer"
        target="_blank"
      >
        title two
      </a>
    </li>
  </ul>
</footer>
`;
